File: go/go_examples/golang-examples-claude-code-deepseek-v4-pro/01_hello_basics/01_hello_world.go
// ============================================================
// 知识点：Hello World — Go 程序的最小结构
//
// 每个 Go 程序由包（package）声明开始。
// package main 定义可独立运行的程序，必须包含 main() 函数作为入口。
// ============================================================

// package 声明：每个 .go 文件都属于某个包
package main

// import 导入标准库或第三方包
// "fmt" 是标准库中最常用的格式化 I/O 包
import "fmt"

// main 函数是程序的入口，自动执行
func main() {
	// Println 在标准输出打印一行文本并自动换行
	fmt.Println("Hello, Go 语言!")

	// Printf 支持格式化占位符
	// %s 表示字符串，%d 表示整数，%v 表示任意类型的值
	name := "Golang"
	version := 1.24
	fmt.Printf("欢迎学习 %s，版本 %.2f\n", name, version)

	// 输出结果：
	// Hello, Go 语言!
	// 欢迎学习 Golang，版本 1.24
}

// 编译运行方法：
// 方法1：go run 01_hello_world.go          ← 直接编译并运行，不生成二进制文件
// 方法2：go build 01_hello_world.go         ← 生成可执行文件 01_hello_world.exe（Windows）
//        ./01_hello_world                   ← 运行生成的可执行文件

File: rust/rust_examples/rust-examples-opencode-mimo-v2.5/01_hello_world/01_hello_world.rs
// ============================================
// 知识点：Hello World 程序
// 难度：入门
// ============================================

// Rust 程序的入口点是 main 函数
// 使用 println! 宏来输出文本到控制台
// 每个 Rust 程序都从 main 函数开始执行

fn main() {
    // println! 是一个宏（注意后面的!），用于格式化输出
    // 它会在输出后自动换行
    println!("Hello, World!");
    
    // 输出空行
    println!();
    
    // 输出带有变量的文本
    let name = "Rust";
    let version = 1.95;
    println!("欢迎学习 {} {} 版本!", name, version);
    
    // 使用格式化占位符
    println!("数字: {}", 42);
    println!("浮点数: {:.2}", 3.14159);
    println!("布尔值: {}", true);
    
    // 多个占位符
    let x = 10;
    let y = 20;
    println!("{} + {} = {}", x, y, x + y);
    
    // 使用命名参数
    println!("{name} 是一种系统编程语言", name = "Rust");
    
    // 输出到标准错误流（不换行）
    eprint!("这是一个错误信息");
    eprintln!("（换行后）");
    
    // 使用 dbg! 宏调试输出（输出到 stderr，包含文件名和行号）
    dbg!(x + y);
}

// ============================================
// 编译和运行方法：
// 
// Windows:
//   rustc 01_hello_world.rs -o 01_hello_world.exe
//   01_hello_world.exe
//
// Linux/macOS:
//   rustc 01_hello_world.rs -o 01_hello_world
//   ./01_hello_world
//
// 或者使用 cargo（推荐）：
//   cargo run --example 01_hello_world
// ============================================
